packages/@dcl/inspector/src/components/ProjectAssetExplorer/ProjectAssetExplorer.tsx

```diff
@@ -1,35 +1,16 @@
-/* eslint-disable no-console */
-import React, { useState } from 'react'
+import { AssetNodeFolder } from './types'
 
-import { AssetNode, AssetNodeFolder } from './types'
-
-import { useSdk } from '../../hooks/sdk/useSdk'
-import { useIsMounted } from '../../hooks/useIsMounted'
-import { DataLayerRpcClient } from '../../tooling-entrypoint'
-import { AssetNodeRootNull, buildAssetTree } from './utils'
+import { useFileSystem } from '../../hooks/catalog/useFileSystem'
+import ProjectView from './ProjectView'
+import { buildAssetTree } from './utils'
 
 import './ProjectAssetExplorer.css'
-import ProjectView from './ProjectView'
 
 export function ProjectAssetExplorer() {
-  const isMounted = useIsMounted()
-  const [_root, setRoot] = useState<AssetNode>(AssetNodeRootNull())
-  const [selectedFolder, setSelectedFolder] = useState<AssetNodeFolder>(AssetNodeRootNull())
-
-  const updateAssetCatalog = async (dataLayer: DataLayerRpcClient) => {
-    const assetcatalog = await dataLayer.getAssetCatalog({})
-    if (!isMounted()) return
-
-    const tree = buildAssetTree(assetcatalog.assets.map((item) => item.path))
-    setRoot(tree)
-    setSelectedFolder(tree)
-  }
-
-  useSdk(async ({ dataLayer }) => {
-    await updateAssetCatalog(dataLayer)
-  })
+  const { files } = useFileSystem()
 
-  const folders = selectedFolder.children.filter((item) => item.type === 'folder') as AssetNodeFolder[]
+  const tree = buildAssetTree(files.assets.map((item) => item.path))
```

Comment:
could this be wrapped in a `useMemo(() => buildAssetTree(...), [files])` so it does not get re-build on every render? Maybe it can be done within a `useAssetTree` hook so it's always done instead of using `buildAssetTree` directly.